Framework/PythonInterface/test/python/mantid/api/SpectrumInfoTest.py

```diff
@@ -2,24 +2,133 @@
+        self.assertEquals(info.size(), 2)
+
+    def test_hasUniqueDetector(self):
+        """ Test if the spectra have unique detectors. """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(info.hasUniqueDetector(0), False)
+        self.assertEquals(info.hasUniqueDetector(1), True)
+
+    def test_l1(self):
+        """ Check if a distance is returned (source to sample). """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.l1()), float)
+
+    def test_l2(self):
+        """ Check if a distance is returned (sample to spectrum). """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.l2(1)), float)
+
+    def test_setMaskedTrue(self):
+        """ Test that the detector's masking can be set to True. """
+        info = self._ws.spectrumInfo()
+        info.setMasked(1, True)
+        self.assertEquals(info.isMasked(1), True)
+
+    def test_setMaskedFalse(self):
+        """ Test that the detector's masking can be set to False. """
+        info = self._ws.spectrumInfo()
+        info.setMasked(1, False)
+        self.assertEquals(info.isMasked(1), False)
+
+    """
+    The following are test cases test for returned V3D objects. The objects
+    represent a vector in 3 dimensions.
+    """
+
+    def test_position(self):
+        """ Test that the spectrum's position is returned. """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.position(1)), V3D)
+
+    def test_sourcePosition(self):
+        """ Test that the source's position is returned. """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.sourcePosition()), V3D)
+
+    def test_samplePosition(self):
+        """ Test that the sample's position is returned. """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.samplePosition()), V3D)
+
+    """
+    The following are test cases are SpectrumDefintion methods.
+    """
+
+    def test_spectrumDefinitionCall(self):
+        """ Get the SpectrumDefintion """
+        info = self._ws.spectrumInfo()
+        self.assertEquals(type(info.spectrumDefinition(1)), SpectrumDefinition)
+
+    def test_spectrumDefintionSize(self):
+        """ Check the size of the SpectrumDefinition
+        (i.e. number of detectors) """
+        info = self._ws.spectrumInfo()
+        spectrumDef = info.spectrumDefinition(1)
+        self.assertEquals(spectrumDef.size(), 1)
+
+    def test_spectrumDefintionAdd(self):
+        """ Add a pair of detector index and time index """
+        info = self._ws.spectrumInfo()
+        spectrumDef = info.spectrumDefinition(1)
+        spectrumDef.add(1, 1)
+        self.assertEquals(spectrumDef.size(), 2)
+
+    def test_spectrumDefintionEquals(self):
+        """ Check the equality of the SpectrumDefintion """
+        info = self._ws.spectrumInfo()
+        spectrumDef = info.spectrumDefinition(1)
+        self.assertTrue(spectrumDef.equals(spectrumDef))
```

Comment:
does `spectrumDef == spectrumDef` work or `spectrumDef != spectrumDef`?

Reply:
Oh I'll check that

Reply:
You do not want `a == b` giving different behaviour from `a.equals(b)` can you imagine what users of your type would think.